### PR TITLE
Pe dg initcond

### PR DESCRIPTION
**Pull request by Paulie14**
- DG interpolation switch (still not sure, no test with significant effect)
- tests 42, 43 for simulation splitting - wiriting and reading init conc in DG

### CODE REVIEW

File: src/transport/assembly_dg.hh [hunk 1]
```diff
@@ -995,23 +995,23 @@ public:
...
     typedef typename TransportDG<Model>::EqFields EqFields;
     typedef typename TransportDG<Model>::EqData EqData;
 
-    static constexpr const char * name() { return "InitConditionAssemblyDG"; }
+    static constexpr const char * name() { return "InitProjectionAssemblyDG"; }
 
     /// Constructor.
-    InitConditionAssemblyDG(EqFields *eq_fields, EqData *eq_data)
+    InitProjectionAssemblyDG(EqFields *eq_fields, EqData *eq_data)
     : AssemblyBase<dim>(eq_data->dg_order), eq_fields_(eq_fields), eq_data_(eq_data) {
```
Comment (jbrezmorf) [inline src/transport/assembly_dg.hh:1008]: @dflanderka Please, remove the order parameter of AssemblyBase as well as creation of default quadrature objects. Move this to the particular assembly classes, take order from eq_data. Some assembly classes doesn't need the harder at all, some need only face/side quadrature...

**Review by jbrezmorf — COMMENTED** [review]
One point for @Paulie14 one for @dflanderka .

File: src/transport/transport_dg.cc [hunk 2]
```diff
@@ -308,12 +311,22 @@ void TransportDG<Model>::initialize()
...
 	stiffness_assembly_ = new GenericAssembly< StiffnessAssemblyDim >(eq_fields_.get(), eq_data_.get());
 	sources_assembly_ = new GenericAssembly< SourcesAssemblyDim >(eq_fields_.get(), eq_data_.get());
 	bdr_cond_assembly_ = new GenericAssembly< BdrConditionAssemblyDim >(eq_fields_.get(), eq_data_.get());
-	init_cond_assembly_ = new GenericAssembly< InitConditionAssemblyDim >(eq_fields_.get(), eq_data_.get());
+    
+    init_proj_assembly_ = nullptr;
+    init_cond_assembly_ = nullptr;
+    if(init_projection)
+	    init_proj_assembly_ = new GenericAssembly< InitProjectionAssemblyDim >(eq_fields_.get(), eq_data_.get());
+    else
```
[the remark this answers is not in the document]
Comment (Paulie14) [reply src/transport/transport_dg.cc:328]: It seems more complicated:
`InitProjectionAssemblyDim` and `InitConditionAssemblyDim` are descendants of AssemblyBase<dim>
however they come into `GenericAssembly` as template parameters and `GenericAssembly` does not have another base class.

Type `GenericAssembly< AssemblyBase >` is not base of `GenericAssembly< InitProjectionAssemblyDim >`
so we cannot do:
```
if(init_projection)
    init_assembly_ = new GenericAssembly< InitProjectionAssemblyDim >(eq_fields_.get(), eq_data_.get());
else
    init_assembly_ = new GenericAssembly< InitConditionAssemblyDim >(eq_fields_.get(), eq_data_.get());
```
Did you have something else in mind?

[the remark this answers is not in the document]
Comment (jbrezmorf) [reply src/transport/transport_dg.cc:328]: I see, please add a GenericAssemblyBase with the virtual method assemble.

**Pull request merged**